runtime/src/main/java/org/corfudb/runtime/object/DiskBackedSMRSnapshot.java

```diff
@@ -48,10 +51,24 @@ public DiskBackedSMRSnapshot(@NonNull OptimisticTransactionDB rocksDb,
         this.set = Collections.newSetFromMap(new WeakHashMap<>());
     }
 
+    private boolean isInvalid() {
+        if (!this.readOptions.isOwningHandle()) {
+            return true; // The snapshot has already been released.
+        }
+
+        if (!this.rocksDb.isOwningHandle()) {
+            log.error("Invalid RocksDB instance {} for snapshot {}.",
+                    rocksDb.getNativeHandle(), version);
+            return true; // RocksDB instance has already been closed.
+        }
+
+        return false;
```

Comment:
The handle can be invalid after the check (race condition). How would this prevent a sigsegv?